### PR TITLE
[Merged by Bors] - Increase durations in tortoise beacon unit test config

### LINKED ISSUE #2582: Block validation failure due to wrong tortoise beacon
## Description

The `TestMultiNode` test may fail because of the `block eligibility check failed - err tortoise beacon eligibility VRF validation failed` error on one of the nodes. This error is logged because this node has a tortoise beacon that is different from tortoise beacons on other nodes. The failure happens almost never on a PC/laptop, sometimes in CI, very often on a Raspberry Pi.

## Steps To Reproduce
1. Run `TestMultiNode` multiple times on a slow machine until it fails

## Expected Behavior

`TestMultiNode` always passes, there are no `block eligibility check failed - err tortoise beacon eligibility VRF validation failed` errors in logs

## Environment
Please complete the following information:
Slow machines like RPi, sometimes CI, but may appear on faster machines.

## Additional Resources 

TBD

### PR DESCRIPTION
**Pull request by nkryuchkov**
## Motivation

Closes #2582

The issue described in #2582 appears because the latest proposal messages, which are sent close to the proposal phase end, had not enough time to be received — they should be received within the grace period, which was `20ms` and not enough for some slow machines. The test config durations were increased.

## Changes

- Increase the tortoise beacon unit test default config durations
- Minor tortoise beacon improvements
- Weak coin fixes

## Test Plan

Unit tests

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

File: tortoisebeacon/tortoise_beacon.go [hunk 1]
```diff
@@ -642,19 +637,23 @@ func (tb *TortoiseBeacon) runConsensusPhase(ctx context.Context, epoch types.Epo
 }
 
 func (tb *TortoiseBeacon) markProposalPhaseFinished(epoch types.EpochID) {
+	finishedAt := time.Now()
+
 	tb.proposalPhaseFinishedTimestampsMu.Lock()
```
Comment (antonlerner) [inline tortoisebeacon/tortoise_beacon.go:642]: can have an RLock here

**Review by antonlerner — APPROVED** [review]

File: tortoisebeacon/tortoise_beacon.go (hunk 1 above)
Comment (nkryuchkov) [reply tortoisebeacon/tortoise_beacon.go:642]: @antonlerner Why? We write here, not read

Comment (antonlerner) [reply tortoisebeacon/tortoise_beacon.go:642]: my bad misread it

**Comment by nkryuchkov** [comment]
bors merge

**Pull request closed without merging**